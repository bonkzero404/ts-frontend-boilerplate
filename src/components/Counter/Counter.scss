@import '../../scss/variable.scss';

.container {
  width: 100%;
  height: 100vh;
}

.box {
  position: absolute;
  top: 50%;
  left: 50%;
  margin-right: -50%;
  transform: translate(-50%, -50%);
  text-align: center;

  a {
    text-decoration: none;
    color: $color-carrot;
    &:hover {
      color: $color-sun;
      text-decoration: none;
    }
  }

  .countText {
    font-family: RobotoRegular;
    color: $color-sassy;
    font-size: 10em;
    text-align: center;
    margin-bottom: 30px;
    line-height: .7em;
  }

  .btnCount {
    font-size: 2em;
    width: 50px;
    height: 50px;
    border-radius: 100%;
    cursor: pointer;
    border: 0px;
    margin-right: 10px;
    margin-left: 10px;
    margin-bottom: 20px;
    color: $color-sassy;
    &:hover {
      color: $color-sun;
    }
  }
}
