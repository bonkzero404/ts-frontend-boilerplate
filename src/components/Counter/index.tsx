import React, { Suspense, FunctionComponent, lazy } from 'react';

const LoadableComponent = lazy(() =>
  import(/* webpackPrefetch: true */ './Counter')
);

const Loading = () => <div>Loading</div>;

const Counter: FunctionComponent<{}> = () => (
  <Suspense fallback={<Loading />}>
    <LoadableComponent />
  </Suspense>
);

export default Counter;
