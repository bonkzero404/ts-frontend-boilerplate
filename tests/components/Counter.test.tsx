import { mountComponent, rendererComponent } from '../utils/test-utils';
import Counter from '../../src/components/Counter/Counter';

describe('<Counter />', () => {
  let wrapper: any;

  const props: any = {};

  it('defines the component', () => {
    wrapper = mountComponent(Counter, props);
    expect(wrapper).toBeDefined();
  });

  it('defines the snapshoot component', () => {
    const tree = rendererComponent(Counter, props).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
