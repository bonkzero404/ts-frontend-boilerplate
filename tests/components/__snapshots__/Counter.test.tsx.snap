// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Counter /> defines the snapshoot component 1`] = `
<div>
  <div>
    <h1>
      0
    </h1>
    <button
      onClick={[Function]}
      type="button"
    >
      +
    </button>
    <button
      onClick={[Function]}
      type="button"
    >
      -
    </button>
    <p>
      Go to 
      <a
        href="/"
        onClick={[Function]}
      >
        Home
      </a>
       for now
    </p>
    <p>
      You are here : 
      /
    </p>
  </div>
</div>
`;
